plotter/field_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter_test
+
+import (
+	"image/png"
+	"log"
+	"os"
+	"testing"
+
+	"gonum.org/v1/plot"
+	"gonum.org/v1/plot/cmpimg"
+	"gonum.org/v1/plot/plotter"
+	"gonum.org/v1/plot/vg"
+	"gonum.org/v1/plot/vg/draw"
+	"gonum.org/v1/plot/vg/vgimg"
+)
+
+type field struct {
+	c, r int
+	fn   func(x, y float64) plotter.XY
+}
+
+func (f field) Dims() (c, r int)           { return f.c, f.r }
+func (f field) Vector(c, r int) plotter.XY { return f.fn(f.X(c), f.Y(r)) }
+func (f field) X(c int) float64 {
+	if c < 0 || c >= f.c {
+		panic("index out of range")
```

Comment:
"x index out of range"

Reply:
This is example code, but OK.

Same in heat_test.go.